rewrite-java/src/main/java/org/openrewrite/java/ReplaceStringLiteralWithConstant.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+
+import java.util.List;
+
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ReplaceStringLiteralWithConstant extends Recipe {
+
+    @Option(displayName = "Fully Qualified Name of the constant",
+            example = "org.springframework.http.MediaType.APPLICATION_JSON_VALUE")
+    String fullyQualifiedNameOfConst;
+
+    @Option(displayName = "(Optional) String literal value that needs to be replaced. Only String literal accepted.",
+            example = "application/json",
+            required = false )
+    @Nullable
+    String literalValue;
+
+    public ReplaceStringLiteralWithConstant(String fullyQualifiedNameOfConst) {
+        this.fullyQualifiedNameOfConst = fullyQualifiedNameOfConst;
+        this.literalValue = fullyQualifiedNameOfConst.substring(fullyQualifiedNameOfConst.lastIndexOf(".")+1);
```

Comment:
This will only work for constants where the name matches the value to be replaced, as seen in the tests for `UTF_8`. But that would not work for the example given on `fullyQualifiedNameOfConst`, where the constant name is `APPLICATION_JSON_VALUE`, but the value to replace is `"application/json"`.

Could this be changed to use the value of the constant, rather than the name of the constant? That way the recipe can be used such as:


```yaml
  - org.openrewrite.java.ReplaceStringLiteralWithConstant:
      fullyQualifiedNameOfConst: org.springframework.http.MediaType.APPLICATION_JSON_VALUE
```